components/button.tsx

```diff
@@ -0,0 +1,46 @@
+import { ReactNode, forwardRef, Ref } from 'react';
+
+type Props = {
+  buttonLink?: boolean;
+  onClick?: () => void;
+  href?: string;
+  disabled?: boolean;
+  otherProps?: unknown;
+  type?: "button" | "submit" | "reset" | undefined;
+  children: string | ReactNode;
+};
+
+type ButtonOrAnchor = Ref<HTMLButtonElement | HTMLAnchorElement>;
+
+const Button = forwardRef<ButtonOrAnchor, Props>(
+  function Button (props, ref) {
+    const { buttonLink, type, children, onClick, href, disabled, ...otherProps } = props;
+    return (
+      <>
+        {buttonLink
+          ? <a {...otherProps} href={href} ref={ref as Ref<HTMLAnchorElement>} onClick={onClick}>{children}</a>
+          : <button type={type || 'button'} onClick={onClick} ref={ref as Ref<HTMLButtonElement>} {...otherProps}>{children}</button>}
+        <style jsx>{`
+          button {
+            cursor: pointer;
+          }
+          a {
+            text-decoration: none;
+            display: inline-block;
+          }
+          a, button {
+            font-size: 26px;
+            line-height: 33px;
+            background: #fff;
+            border: 2px solid ${disabled ? '#b0b0b0' : '#222222'};
+            color: ${disabled ? '#b0b0b0' : '#222222'};
+            padding: 10px 20px;
+            border-radius: 50px;
+          }
+        `}
+        </style>
+      </>
+    );
+});
+
+export default Button;
```

Comment:
Default exports are sometimes a little weird/fraught (different way of importing versus named imports). I don't have a strong opinion here, but a Mux style guide that either defines when they should be used, or stating that they should not be used, might be worthwhile.

Reply:
I can't think of a time that I've seen React components export anything other than default.

For other modules, yes. But for a single file with 1 React component I think `export default` is what we want so then we can do

```ts
import Button from './button'
```

Reply:
That always makes me uncomfortable because of the pattern of sticking HOCs in the same file as their base component, but I don't feel strongly about it.